src/happy.rs

```diff
@@ -0,0 +1,31 @@
+use std::str::FromStr;
+
+use hickory_client::{
+    client::{Client, SyncClient},
+    error::ClientResult,
+    op::DnsResponse,
+    rr::{DNSClass, Name, RecordType},
+    udp::UdpClientConnection,
+};
+
+/*
+TODO:
+This needs to be TCP and have Timeout Retries
+
+*/
+pub fn search(identifier: Name, rtype: RecordType) -> anyhow::Result<DnsResponse> {
+    debug!("searching {:?} for {}", rtype, identifier);
+    let socket_addr = std::net::SocketAddr::new(std::net::IpAddr::from_str("1.1.1.1")?, 53);
+    let client = happy_eyeballs::connect(socket_addr)?;
+    let questionable_addr = client.peer_addr()?;
```

Comment:
I'm not quite sure anymore, why are we doing happy eyeballs here at all? From the config, there is no doubt on whether to use IPv4 or IPv6, as we're specifying IPs and not domain names. The `questionable_addr` extracted from the eyeballed connection isn't used either. Maybe we should drop those bits completely.

Reply:
I removed it while debugging and didn't notice when committing- good catch.
Given we parse only a single resolver in the config it should not make much of a difference anyway.